internal/pipeline/types_test.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pipeline_test
+
+import (
+	"reflect"
+	"testing"
+
+	. "github.com/GoogleContainerTools/kpt/internal/pipeline"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func parseFromString(input string) (Pipeline, error) {
+	var p Pipeline
+	err := yaml.Unmarshal([]byte(input), &p)
+	if err != nil {
+		return p, err
+	}
+	return p, nil
+}
+
+func isFunctionEqual(t *testing.T, f1, f2 Function) bool {
```

Comment:
Try to keep the helper methods at the bottom of the file to improve the readability.